tests/components/environment_canada/test_weather.py

```diff
@@ -15,14 +17,24 @@
 from tests.common import load_fixture
 
 
+def date_hook(weather):
+    """Convert timestamp string to datetime."""
+
+    if t := weather.get("timestamp"):
+        with contextlib.suppress(ValueError):
+            weather["timestamp"] = datetime.fromisoformat(t)
+    return weather
+
+
 async def test_forecast_daily(
     hass: HomeAssistant,
     snapshot: SnapshotAssertion,
 ) -> None:
     """Test basic forecast."""
 
     ec_data = json.loads(
-        load_fixture("environment_canada/current_conditions_data.json")
+        load_fixture("environment_canada/current_conditions_data.json"),
+        object_hook=date_hook,
```

Comment:
This doesn't seem right. If it's coming from the api as a string we should in that case convert it in the `weather` platform. Add the proper timezone and then convert it to an isoformatted datetime string.

Reply:
Would probably have been easier if you kept it as a string in the lib but it's what it is (as I took a look and notice that you're converting it).

So I guess we need to have this conversion here, maybe we can move it to be a fixture in `conftest.py` instead as it's also used in `test_diagnostics.py`